### DISCUSSION TITLE
useGetIdentity doesn't get identity 😥

### DISCUSSION THREAD

**Discussion by LiaCiobanu001** [refinedev/refine#4459] [Q&A]
import { useGetIdentity } from "@refinedev/core";

const { data: user } = useGetIdentity({
        v3LegacyAuthProviderCompatible: true,
    });
    const [propertyImage, setPropertyImage] = useState({ name: "", url: "" });
    const {
        refineCore: { onFinish, formLoading },
        register,
        handleSubmit,
    } = useForm();

    const handleImageChange = (file: File) => {
        const reader = (readFile: File) =>
            new Promise<string>((resolve, reject) => {
                const fileReader = new FileReader();
                fileReader.onload = () => resolve(fileReader.result as string);
                fileReader.readAsDataURL(readFile);
            });

        reader(file).then((result: string) =>
            setPropertyImage({ name: file?.name, url: result }),
        );
    };

    const onFinishHandler = async (data: FieldValues) => {
    if (!propertyImage.name) return alert("Please select an image");

    if (!user || !user.email) {
        return alert("User email is not available");
    }

    await onFinish({
        ...data,
        photo: propertyImage.url,
        email: user.email,
    });
};

Soooo, It keeps giving me the error " User email is not available ", everything is ok with the code, I also imported everything, according to my searches on the net, the problem would be in "useGetIdentity  ". I also looked over the documentation and nothing seems wrong.
Maybe it is imported differently or this import is used in a different way, please help me a little.

**Comment by yildirayunlu** [comment]
Hey @LiaCiobanu001 I didn't fully understand what the problem was. Does authentication work properly? Or do you want to update the data of the logged in user?

I recommend you to review the [Auth Provider](https://refine.dev/docs/api-reference/core/providers/auth-provider/) section.

**Reply by LiaCiobanu001** [reply]
the authentication works ok, everything is also registered in mongodb, I think the problem is really in the Auth Provider. I will take a look at the link you sent me. Thanks a lot for your help ❤️ 
If we are still here 😄 
How do I get rid of this thing? 
![Screenshot 2023-06-07 171513](https://github.com/refinedev/refine/assets/101062343/26316ed5-683a-40c0-b042-6c1fa654aee8)
